g2o/types/slam3d/edge_se3_prior.cpp

```diff
@@ -72,10 +72,6 @@ namespace g2o {
         if (i!=j)
           information()(j,i)=information()(i,j);
       }
-    if (is.bad()) {
-      //  we overwrite the information matrix
-      information().setIdentity();
```

Comment:
Seems like a bad change, It means if you only read part of your transform you'd treat it as valid and use the partially initialized transform as is.

Reply:
Oh since the information matrix was already optional I guess it was already an issue

Reply:
Although if there's a partial read then it should probably still zero out the information matrix

Reply:
Issue #162 is fixed by the other commit.

If there is an I/O error that leads to the bad bit being set, resetting the information matrix to an identity matrix will change the optimisation result and there is no way for me to know that the result is not the expected result. Therefore it is potentially harmful.
A safer way to deal with this type of issue would be to ensure that either no matrix is read (the identity being the default in this case) or a full matrix is read, and to raise an error otherwise.
I just noticed that the same fallback also exists in EdgeSE3PointXYZ.

I still think it's potentially harmful, but if you think it's the best way of handling this, I can issue another pull request with just the other commit.

Reply:
I have now added another commit which implements the failure behaviour that I suggested above. Method read now returns true if either the whole matrix is read or no matrix is given and false otherwise.